st.js

```diff
@@ -243,73 +250,94 @@ Mount.prototype.serve = function (req, res, next) {
     return true
   }
 
-  // now we have a path.  check for the fd.
-  this.cache.fd.get(p, function (er, fd) {
-    // inability to open is some kind of error, probably 404
-    // if we're in passthrough, AND got a next function, we can
-    // fall through to that.  otherwise, we already returned true,
-    // send an error.
-    if (er) {
-      if (this.opt.passthrough === true && er.code === 'ENOENT' && next)
-        return next()
-      return this.error(er, res)
-    }
+  var exts = this.extensions
+  var attempt = 0
 
-    // we may be about to use this, so don't let it be closed by cache purge
-    this.fdman.checkout(p, fd)
-    // a safe end() function that can be called multiple times but
-    // only perform a single checkin
-    var end = this.fdman.checkinfn(p, fd)
+  // when using fallback extensions, cache the result of
+  // any matched URLs.  otherwise, we'd end up requesting
+  // missing files from the fd cache on each request.
+  if (this.extensions && this.pathMap[p])
```

Comment:
seems like a risk of stale lookups if the files are changed underneath (`mv index.htm index.html`); this may need to be better integrated to the cache to deal with conflicts I suspect